pkg/provisioner/controller.go

```diff
@@ -638,3 +656,48 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucketClaim, class *storagev1.StorageClass) error {
+	ob, err := getObjectBucketFromClaimKey(key, obc, c.libClientset, defaultRetryBaseInterval, defaultRetryTimeout)
+	if err != nil {
+		log.Error(err, "getting ob failed", "key", key)
+		return err
+	}
+	if reflect.DeepEqual(ob.Spec.Endpoint.AdditionalConfigData, obc.Spec.AdditionalConfig) {
+		return nil
+	} else {
+		logD.Info("Additional config changed, hence updating OB")
+		tmp := make(map[string]string)
+
+		copyAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData, tmp)
+		clearAdditionalConfig(ob.Spec.Endpoint.AdditionalConfigData)
+		copyAdditionalConfig(obc.Spec.AdditionalConfig, ob.Spec.Endpoint.AdditionalConfigData)
+
+		err = c.provisioner.Update(ob)
```

Comment:
I know an earlier version of this pr passed an options struct to Update and now you're passing the entire OB, with the need to save a tmp copy of additionalConfig, etc. What is the reason for not just passing to Update a single parameter named `additionalConfig` as a map[string]string? Then provisioners don't need to worry about what else may have changed in the OB. It feels safer and simpler, but at the cost of being less future-proof if it's decided to support more OB mutations.

Reply:
@jeffvance : If I pass only the `additionalconfig` the provisioned does not much idea about details like which user or bucket(which might be opaque to library and different for each) it is referring to. Easier way to pass `OB` information like `Delete()` or `Revoke()`. Or I need to figure what all must is needed(from my point of view I can only check from Rook)